### PR TITLE
Add support for HuggingFace's TensorFlow models

### PR DESCRIPTION
**Pull request by valeriosofi**
**New Feature:**

This PR implements support for HuggingFace's TensorFlow models, solving #125.
It also adds a notebook that shows a use case with BERT.

Example of usage:
```python
from speedster import optimize_model
from transformers import TFAutoModel, AutoTokenizer

# Make sure to always load a matching tokenizer and model!
tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
model = TFAutoModel.from_pretrained("bert-base-cased")

# Let's load some data and tokenize it
test_string = "This is a sentence!"
tokenized_inputs = tokenizer(test_string, return_tensors="np")

dynamic_info = {
    "inputs": [
        {0: "batch", 1: "num_tokens"},
        {0: "batch", 1: "num_tokens"},
        {0: "batch", 1: "num_tokens"},
    ],
    "outputs": [{0: "batch", 1: "num_tokens"}, {0: "batch"}],
}

# Run Speedster optimization
optimized_model = optimize_model(
  model, input_data=[tokenized_inputs], dynamic_info=dynamic_info
)
```

**Minor changes:**
- Added a warning when dynamic_info is not provided with HuggingFace models
- The final report on the optimized model now contains also the name of the compiler used
- Fix a tvm issue when using the PyTorch implementation with gpu
- Added dynamic shape support for optimized models converted from TensorFlow to ONNX
- Fixed an issue that caused a wrong output shape for the models optimized with the neural compressor
- Enabled openVINO compiler when using python 3.10

### CODE REVIEW

**Review by nebuly-ai — APPROVED** [review]
LGTM!

File: apps/accelerate/speedster/speedster/api/tests/test_tensorflow.py [hunk 1]
```diff
@@ -23,7 +22,7 @@
     TensorflowApacheTVMInferenceLearner,
 )
 from nebullvm.operations.optimizations.compilers.utils import tvm_is_available
-from nebullvm.tools.utils import is_python_version_3_10
+from nebullvm.tools.utils import gpu_is_available
 from speedster import optimize_model
```
Comment (diegofiori) [inline apps/accelerate/speedster/speedster/api/tests/test_tensorflow.py:26]: in speedster nebullvm should be imported with the other third party libraries

File: nebullvm/operations/inference_learners/neural_compressor.py [hunk 2]
```diff
@@ -41,6 +41,8 @@ class NeuralCompressorInferenceLearner(BaseInferenceLearner, ABC):
         model (torch.fx.GraphModule): Torch fx graph model.
     """
 
+    name = "IntelNeuralCompressor"
```
Comment (diegofiori) [inline nebullvm/operations/inference_learners/neural_compressor.py:44]: Just a question: where do we use the name?

**Comment by diegofiori** [comment]
@valeriosofi why was the azure test cancelled?

File: nebullvm/operations/inference_learners/neural_compressor.py (hunk 2 above)
Comment (valeriosofi) [reply nebullvm/operations/inference_learners/neural_compressor.py:44]: It is printed in the final report, in order to print the name of the technique used to generate the optimized model.

File: apps/accelerate/speedster/requirements.txt [hunk 3]
```diff
@@ -1 +1,2 @@
 nebullvm>=0.6.1
+tabulate
```
Comment (diegofiori) [inline apps/accelerate/speedster/requirements.txt:2]: please specify a minimum version

File: apps/accelerate/speedster/setup.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 REQUIREMENTS = [
     "nebullvm>=0.6.1",
+    "tabulate",
```
Comment (diegofiori) [inline apps/accelerate/speedster/setup.py:7]: same as before

**Comment by valeriosofi** [comment]
> @valeriosofi why was the azure test cancelled?

Because there was a bug, so there was no reason to perform the tests, they would have failed

File: apps/accelerate/speedster/speedster/root_op.py [hunk 5]
```diff
@@ -13,6 +13,8 @@
     List,
 )
 
+from tabulate import tabulate
```
Comment (diegofiori) [inline apps/accelerate/speedster/speedster/root_op.py:16]: tabulate should be imported in the same level as nebullvm

**Comment by diegofiori** [comment]
Closes #125 and #103

**Pull request merged**